### PR TITLE
feat(node): add `public_host` and `public_port` to config

### LINKED ISSUE #1002: Add `public_hostname` and `public_port` to node config file and use this also for the TLS server name
This can be used for the `register` command and also to update the node's data as part of #965.

### PR DESCRIPTION
**Pull request by mlegner**
Closes #1002

### CODE REVIEW

File: crates/walrus-service/bin/node.rs [hunk 1]
```diff
@@ -195,9 +189,12 @@ struct ConfigArgs {
     /// Initial storage capacity of this node in bytes.
     ///
     /// The value can either by unitless; have suffixes for powers of 1000, such as (B),
-    /// kilobytes (K), etc, or have suffixes for the IEC units such as kibibytes (Ki),
+    /// kilobytes (K), etc.; or have suffixes for the IEC units such as kibibytes (Ki),
     /// mebibytes (Mi), etc.
     node_capacity: ByteCount,
+    #[clap(long)]
+    /// The host name or public IP address of the node.
+    public_host: String,
```
Comment (halfprice) [inline crates/walrus-service/bin/node.rs:206]: Question: what's the relationship between this in IP format and rest_api_address? Seem that they should match?

File: crates/walrus-service/bin/node.rs [hunk 2]
```diff
@@ -483,37 +475,34 @@ mod commands {
...
+        let public_port = config.public_port.unwrap_or(REST_API_PORT);
         let public_address = if let Ok(ip_addr) = IpAddr::from_str(&public_host) {
-            NetworkAddress(SocketAddr::new(ip_addr, port).to_string())
+            NetworkAddress(SocketAddr::new(ip_addr, public_port).to_string())
         } else {
-            // Do a minor sanity check that the user has not included a port in the hostname
-            ensure!(
-                !public_host.contains(':'),
-                "DNS names must not contain ':', to specify a port, use the --port option."
-            );
```
Comment (halfprice) [inline crates/walrus-service/bin/node.rs:520]: Nit: this check seems still useful

File: crates/walrus-service/bin/node.rs (hunk 1 above)
Comment (mlegner) [reply crates/walrus-service/bin/node.rs:206]: The `rest_api_address` can be something like `0.0.0.0` or `127.0.0.1`. The `public_host` should be the public IP address of the node or any reverse proxy in front of it.

File: crates/walrus-service/bin/node.rs (hunk 2 above)
Comment (mlegner) [reply crates/walrus-service/bin/node.rs:520]: This is now included in the `generate_config` function. But I can keep this here as well.

**Pull request merged**